### PR TITLE
sort imports using LocalPrefix

### PR DESCRIPTION
**Pull request by Thearas**
Implement https://github.com/ent/ent/issues/2170.

The second commit is generated by `go generate ./...`.

### CODE REVIEW

File: entc/gen/graph.go [hunk 1]
```diff
@@ -751,7 +751,9 @@ func (a assets) write() error {
 }
 
 // format runs "goimports" on all assets.
-func (a assets) format() error {
+func (a assets) format(localPrefix string) error {
+	imports.LocalPrefix = localPrefix
+
```
Comment (masseelch) [inline entc/gen/graph.go:756]:
```suggestion
```

**Review by masseelch — APPROVED** [review]
I like very much. Let's see what @a8m thinks.

**Comment by a8m** [comment]
> I like very much. Let's see what @a8m thinks.

Thanks for your contribution @Thearas, and for your review @masseelch ❤️ 

Please, see my comment on https://github.com/ent/ent/issues/2170#issuecomment-980009054, and let's continue the discussion there.

File: entc/integration/gremlin/ent/fieldtype.go [hunk 2]
```diff
@@ -17,8 +17,9 @@ import (
 	"entgo.io/ent/dialect/gremlin"
 	"entgo.io/ent/entc/integration/ent/role"
 	"entgo.io/ent/entc/integration/ent/schema"
-	"entgo.io/ent/entc/integration/gremlin/ent/fieldtype"
 	"github.com/google/uuid"
+
+	"entgo.io/ent/entc/integration/gremlin/ent/fieldtype"
```
Comment (a8m) [inline entc/integration/gremlin/ent/fieldtype.go:22]: In such cases, the generated package imports a "local package". Therefore, the grouping should be based on the Go modules (project root). In this case, it's `entgo.io/ent`. i.e. if we have 2 import paths:
```go
<root-pkg>/ent/user
<root-pkg>/pkg/types
```

We want to set `-local` to `<root-pkg>`, and not `<root-pkg>/ent`, right?

Comment (masseelch) [reply entc/integration/gremlin/ent/fieldtype.go:22]: Ahm, yeah. `goimports` needs an option to add a 4th section :-D

Comment (Thearas) [reply entc/integration/gremlin/ent/fieldtype.go:22]: Yeah, I agree😁. How to get the `<root-pkg>`?

Comment (Thearas) [reply entc/integration/gremlin/ent/fieldtype.go:22]: Another idea, `entc` will set `LocalPrefix` to `<root-pkg>` only if the `LocalPrefix` is empty.

Comment (a8m) [reply entc/integration/gremlin/ent/fieldtype.go:22]: We can add some code for loading the Go module information (see my PR - https://github.com/ent/ent/pull/2175) and use it as a LocalPrefix, but I agree it should be set only if `LocalPrefix` is empty. Let me know what do you think.

Comment (a8m) [reply entc/integration/gremlin/ent/fieldtype.go:22]: `Module.Path` should return the go module path. But again, tbh, I think there are no real guidelines on how package paths should be grouped. I, for example, like to group all my company/product import paths in the same block. Thoughts?

Comment (Thearas) [reply entc/integration/gremlin/ent/fieldtype.go:22]: I agree. I have already fixed my problem with `imports.LocalPrefix = "<config-goes-here>"`. This PR could be closed😂.

Comment (Thearas) [reply entc/integration/gremlin/ent/fieldtype.go:22]: Thanks for your solution @a8m !

**Pull request closed without merging**